crates/ruff_linter/resources/test/fixtures/ruff/for_variable_used_after_block.py

```diff
@@ -0,0 +1,81 @@
+for global_var in []:
+    _ = global_var
+    pass
+
+# ❌
+_ = global_var
+
+def foo():
+    # For control var used outside block
+    for event in []:
+        _ = event
+        pass
+
+    # ❌
+    _ = event
+
+    for x in []:
+        _ = x
+        pass
+    
+    # Using the same control variable name in a different loop is ok
+    for x in []:
+        _ = x
+        pass
+
+    for y in []:
+        # ❌ x is used outside of the loop it was defined in (meant to use y)
+        if x == 5:
+            pass
+
+    # Assign a variable before the loop
+    room_id = 3
+    _ = room_id
+    # Use the same variable name in a loop
+    for room_id in []:
+        _ = room_id
+        pass
+
+    # ❌ After the loop is not ok because the value is probably not what you expect
+    _ = room_id
+
+    # ❌ Augmented assignment is not allowed because the value is probably not what you expect
+    room_id += 1
+
+    # Assigning again after the loop is ok
+    room_id = 5
+    room_id += 1
+    _ = room_id
+
```

Comment:
#### Multiple usages in the same statement

> It would be great if we only flagged each variable once (at least per statement?)
> 
> * [src/_pytest/assertion/rewrite.py:492:40:](https://github.com/pytest-dev/pytest/blob/ef9b8f9d748b6f50eab5d43e32d93008f7880899/src/_pytest/assertion/rewrite.py#L492) RUF032 `for` loop variable i is used outside of block
> * [src/_pytest/assertion/rewrite.py:492:53:](https://github.com/pytest-dev/pytest/blob/ef9b8f9d748b6f50eab5d43e32d93008f7880899/src/_pytest/assertion/rewrite.py#L492) RUF032 `for` loop variable i is used outside of block
> * [src/_pytest/assertion/rewrite.py:492:66:](https://github.com/pytest-dev/pytest/blob/ef9b8f9d748b6f50eab5d43e32d93008f7880899/src/_pytest/assertion/rewrite.py#L492) RUF032 `for` loop variable i is used outside of block

Seems like a decent improvement. Do we do this sort of deduplication for other rules? For the violation `Diagnostic`, do we combine the `TextRange` to cover the whole area?